### PR TITLE
ODC-6783: Export perspectives to console-config ConfigMap

### PR DESCRIPTION
**Pull request by debsmita1**
**Story:**
https://issues.redhat.com/browse/ODC-6783

Add dev catalog types customization API based on https://github.com/openshift/enhancements/pull/1205

Test setup:
This PR bumps the openshift/api, includes changes from https://github.com/openshift/api/pull/1269, and finally copies the data from the cluster Console configuration to the console-config ConfigMap.

To test this change launch a cluster with launch https://github.com/openshift/console-operator/pull/678

Open https://cluster/k8s/cluster/operator.openshift.io~v1~Console/cluster/yaml and configure customization like:
```
  customization:
    perspectives:
    - id: admin
       visibility:
         state: AccessReview
         accessReview:
         - resource: namespaces
            verb: list
    - id: custom-admin
      visibility:
         state: Disabled
```
Open https://cluster/k8s/ns/openshift-console/configmaps/console-config and check the data key console-config.yaml. The YAML should contain the perspectives after a few seconds.

```
customization:
    perspectives:
    - id: admin
       visibility:
         state: AccessReview
         accessReview:
         - resource: namespaces
            verb: list
    - id: custom-admin
      visibility:
         state: Disabled
```

### CODE REVIEW

**Comment by invincibleJai** [comment]
looks like it needs rebase

**Comment by debsmita1** [comment]
changing it to WIP as the API is not merged yet

File: pkg/console/subresource/consoleserver/types.go [hunk 1]
```diff
@@ -96,6 +97,8 @@ type Customization struct {
 	QuickStarts      QuickStarts                           `yaml:"quickStarts,omitempty"`
 	// addPage allows customizing actions on the Add page in developer perspective.
 	AddPage AddPage `yaml:"addPage,omitempty"`
+	// perspectives allows enabling/disabling of perspective(s) that user can see in the Perspective switcher dropdown.
+	Perspectives []Perspectives `yaml:"perspectives,omitempty"`
```
Comment (jerolimov) [inline pkg/console/subresource/consoleserver/types.go:101]:
```suggestion
	Perspectives []Perspective `yaml:"perspectives,omitempty"`
```

File: pkg/console/subresource/consoleserver/types.go [hunk 2]
```diff
@@ -141,6 +144,48 @@ type AddPage struct {
...
+type PerspectiveVisibility struct {
+	// state defines the perspective is enabled or disabled or access review check is required.
+	// state is required
+	State PerspectiveState `yaml:"state"`
+	// accessReview defines required and missing access review checks.
+	AccessReview *PerspectiveAccessReview `yaml:"accessReview,omitempty"`
+}
+
+// Perspectives allows enabling/disabling of perspective(s).
+type Perspectives struct {
```
Comment (jerolimov) [inline pkg/console/subresource/consoleserver/types.go:178]:
```suggestion
// Perspective allows enabling/disabling of perspective(s).
type Perspective struct {
```

**Review by jerolimov — CHANGES_REQUESTED** [review]
Can we use singular here as well

File: go.mod [hunk 3]
```diff
@@ -101,16 +101,11 @@ require (
...
 	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
 	sigs.k8s.io/kube-storage-version-migrator v0.0.4 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
-
-replace (
-	github.com/openshift/api => github.com/jhadvig/api v0.0.0-20220909092727-aab1693447da
-	github.com/openshift/client-go => github.com/jhadvig/client-go v0.0.0-20220909094556-07a1d60cb20c
-)
```
Comment (jerolimov) [inline go.mod:116]: :+1:

**Review by jerolimov — APPROVED** [review]
Tested this successfully on a cluster bot instance, together with https://github.com/openshift/console/pull/12075 :+1: 

Console ConfigMap was updated correctly.

/lgtm
/label qe-approved

**Comment by jerolimov** [comment]
propagate from the Epic:
/label px-approved
/label docs-approved

/retest

**Comment by jerolimov** [comment]
/retitle ODC-6783: Export perspectives to console-config ConfigMap

**Review by jhadvig — APPROVED** [review]
/approve

**Pull request merged**